Update downloading-workflow-artifacts.md

TheNH813 opened the pull request:
Artifacts cannot be downloaded unless a user is logged in, and this is not specified in the current version of the page. Updating for clarity.

<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to an issue) and what changes you've made, we can triage your pull request to the best possible team for review.
-->

### Why:

It was unclear why I was not able to download artifacts for a project I wanted to help test. After doing some research, I concluded that even for public repositories, users must be logged in to access artifacts. Even for public repositories, the downloads links are not there unless a user is logged in. This information should be in the relevant documentation page so it is clear to users that this is a requirement.

<!-- If there's an existing issue for your change, please link to it in the brackets above.
If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Added statement that artifacts cannot be downloaded unless a user is logged in.

<!-- Let us know what you are changing. Share anything that could provide the most context.
If you made changes to the `content` directory, a table will populate in a comment below with links to the preview and current production articles. -->

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

TheNH813 committed:
Update downloading-workflow-artifacts.md

Artifacts cannot be downloaded unless a user is logged in, and this is not specified in the current version of the page. Updating for clarity.

cmwilson21 commented:
@TheNH813 Thanks so much for opening a PR! I'll get this triaged for review :zap:

TheNH813 committed:
Add "permissions:" statement for artifacts

Reviewed contribution documentation, examples and localization checklist for wording, and adjusted suggestion to use a "permissions:" statement.

TheNH813 reviewed (commented):
@lecoursen
Made the requested changes to suggestion.

TheNH813 committed:
Merge branch 'main' into patch-3

lecoursen commented on content/actions/managing-workflow-runs/downloading-workflow-artifacts.md:
```diff
@@ -1,6 +1,7 @@
 ---
 title: Downloading workflow artifacts
 intro: You can download archived artifacts before they automatically expire.
+permissions: Authorized accounts with read access to a repository can download artifacts.
```
Thanks so much for making that change @TheNH813! I'm going to tweak it just slightly to really reinforce the fact that you need to be signed in as you so helpfully pointed out. 💖 

```suggestion
permissions: People who are signed into {% data variables.product.product_name %} and have read access to a repository can download workflow artifacts.
```

lecoursen committed:
Add :nail_care:

lecoursen committed:
Merge branch 'main' into patch-3

lecoursen committed:
Merge branch 'main' into patch-3

lecoursen committed:
Merge branch 'main' into patch-3

violethaze74 reviewed (commented):
files

lecoursen committed:
Merge branch 'main' into patch-3
